int32 face_x
int32 face_y